src/components/header.tsx

```diff
@@ -0,0 +1,226 @@
+                          asChild>
+                          <a className="flex h-full w-full select-none flex-col justify-end rounded-md p-6 no-underline outline-none focus:shadow-md">
+                            <div className="mb-2 mt-4 text-lg font-medium text-white">
+                              {headerSection.colorSubMenu.subHeader}
+                            </div>
+                            <p className="text-sm leading-tight text-white">
+                              {headerSection.colorSubMenu.text}
+                            </p>
+                          </a>
+                        </NavigationMenuLink>
+                      </li>
+                      {headerSection.items.map((item, key) => (
+                        <ListItem
+                          key={key}
+                          href={item.link}
+                          title={item.subHeader}>
+                          {item.text}
+                        </ListItem>
+                      ))}
+                    </ul>
+                  </NavigationMenuContent>
+                </NavigationMenuItem>
+              ))}
+              <NavigationMenuItem>
+                <Link href="/news" legacyBehavior passHref>
+                  <NavigationMenuLink className={navigationMenuTriggerStyle()}>
+                    ČLÁNKY
+                  </NavigationMenuLink>
+                </Link>
+              </NavigationMenuItem>
+            </NavigationMenuList>
+          </NavigationMenu>
+        </div>
+
+        <div className="flex gap-3">
+          <div className="px-6 hidden sm:block">
+            <Button>Prihlásenie</Button>
+          </div>
+          <Button variant="outline" size="icon">
+            <HelpOutlineIcon />
+            <span className="sr-only">Support</span>
+          </Button>
+          <Button variant="outline" size="icon">
+            <LanguageIcon />
+            <span className="sr-only">Choose Language</span>
+          </Button>
+
+          <Sheet>
+            <SheetTrigger asChild>
+              <Button className="lg:hidden" variant="outline" size="icon">
+                <MenuIcon />
+                <span className="sr-only">Toggle menu</span>
+              </Button>
+            </SheetTrigger>
+            <SheetContent>
+              <div className="grid gap-4 p-4">
+                {hamburgerList.map((link, key) => (
+                  <Link
+                    key={key}
+                    className="inline-flex h-9 items-center justify-center rounded-md bg-gray-100 px-4 py-2 text-sm font-medium transition-colors hover:bg-gray-100/50 focus:bg-gray-100/50 dark:bg-gray-800 dark:hover:bg-gray-800/50 dark:focus:bg-gray-800/50"
+                    href={link.link}>
+                    {link.text}
+                  </Link>
+                ))}
+              </div>
+            </SheetContent>
+          </Sheet>
+        </div>
+      </div>
+    </nav>
+  );
+}
+
+const ListItem = React.forwardRef<
```

Comment:
Toto si asi nevytvoril ty tipujem :D Väčšinou takéto veci môžu byť kludne spolu, ale tým že už na prvý pohľad to vyzerá zložito by som to vytiahol do samostatného súboru. Rovnako sa snaž oddelovať vygenerované veci zo shadcn a veci ktoré si spravil ty sám (shadcn nechávaj v tom ui folderi, zvyšok inam).

Reply:
done